crates/meroctl/src/connection.rs

```diff
@@ -1,26 +1,36 @@
-use chrono::Utc;
 use eyre::{bail, eyre, Result as EyreResult};
-use libp2p::identity::Keypair;
 use reqwest::Client;
 use serde::de::DeserializeOwned;
 use serde::Serialize;
 use url::Url;
 
+use crate::cli::storage::get_storage;
 use crate::cli::ApiError;
 use crate::common::RequestType;
 
+#[derive(Debug, thiserror::Error)]
+pub enum TokenError {
+    #[error("Token refresh failed")]
+    RefreshFailed,
+    #[error("Token expired and refresh token is not available")]
+    NoRefreshToken,
+    #[error("Failed to store refreshed token: {0}")]
+    StorageError(#[from] eyre::Error),
+}
```

Comment:
you should use `eyre::Report`, btw if you didn't know, errors like these don't embed backtrace information, that's somewhat tolerable in libraries but CLIs are end apps, `eyre`'s facilities are preferable